iceoryx2-bb/container/src/byte_string.rs

```diff
@@ -237,6 +237,31 @@ impl<const CAPACITY: usize> FixedSizeByteString<CAPACITY> {
         Ok(new_self)
     }
 
+    /// Creates a new [`FixedSizeByteString`] from a byte slice. If the byte slice does not fit
+    /// into the [`FixedSizeByteString`] it will be truncated.
+    pub fn from_bytes_truncated(bytes: &[u8]) -> Self {
+        let mut new_self = Self::new();
+        new_self.len = std::cmp::min(bytes.len(), CAPACITY);
+        unsafe {
+            std::ptr::copy(
+                bytes.as_ptr(),
+                new_self.data.as_ptr() as *mut u8,
+                bytes.len(),
+            )
+        };
```

Comment:
I think this can be condensed to `new_self.data.copy_from_slice(&bytes[0..new_self.len])`. No need to use unsafe.

Edit: oh, wait. It's a `MaybeUninit`. In this case you can use `new_self.data.write(...)`